plugins/package-managers/bitbake/src/funTest/kotlin/BitBakeToolFunTest.kt

```diff
@@ -0,0 +1,102 @@
+
+import org.ossreviewtoolkit.analyzer.Analyzer
+import org.ossreviewtoolkit.analyzer.create
+import org.ossreviewtoolkit.model.Identifier
+import org.ossreviewtoolkit.model.VcsInfo
+import org.ossreviewtoolkit.model.VcsType
+import org.ossreviewtoolkit.model.config.AnalyzerConfiguration
+import org.ossreviewtoolkit.plugins.versioncontrolsystems.git.Git
+import org.ossreviewtoolkit.utils.test.ExpensiveTag
+import org.ossreviewtoolkit.utils.test.shouldNotBeNull
+
+class BitBakeToolFunTest : WordSpec({
+    "BitBake" should {
+        "get the version correctly" {
+            val bitBake = create("BitBake") as BitBake
+
+            val version = bitBake.getBitBakeVersion(tempdir())
+
+            version shouldMatch "\\d+\\.\\d+\\.\\d+"
+        }
+    }
+
+    "Analyzing recipes from Poky" should {
+        val projectDir = tempdir()
+        val pokyVcsInfo = VcsInfo(VcsType.GIT, "https://git.yoctoproject.org/poky", "kirkstone-4.0.17")
+
+        Git().run {
+            val workingTree = initWorkingTree(projectDir, pokyVcsInfo)
+            updateWorkingTree(workingTree, pokyVcsInfo.revision)
+        } shouldBeSuccess pokyVcsInfo.revision
+
+        "create an SPDX file for the 'quilt-native' package" {
+            val recipeFileName = "quilt-native_0.67.bb"
+            val result = Analyzer(AnalyzerConfiguration()).run {
+                val fileInfo = findManagedFiles(projectDir)
+                val singleFileInfo = fileInfo.copy(
+                    managedFiles = fileInfo.managedFiles.map { (packageManager, definitionsFiles) ->
+                        packageManager to definitionsFiles.filter { it.name == recipeFileName }
+                    }.toMap()
+                )
+                analyze(singleFileInfo)
+            }
+
+            result.analyzer?.result shouldNotBeNull {
+                projects shouldHaveSize 1
+
+                with(projects.single()) {
+                    id shouldBe Identifier("BitBake:OpenEmbedded ():quilt-native:0.67")
+                    declaredLicenses shouldBe setOf("GPL-2.0-only")
+                    homepageUrl shouldBe "http://savannah.nongnu.org/projects/quilt/"
+                    scopes should beEmpty()
+                }
+            }
+        }
+
+        "create a SPDX files for the 'xmlto' package".config(tags = setOf(ExpensiveTag)) {
+            val recipeFileName = "xmlto_0.0.28.bb"
+            val result = Analyzer(AnalyzerConfiguration()).run {
+                val fileInfo = findManagedFiles(projectDir)
+                val singleFileInfo = fileInfo.copy(
+                    managedFiles = fileInfo.managedFiles.map { (packageManager, definitionsFiles) ->
+                        packageManager to definitionsFiles.filter { it.name == recipeFileName }
+                    }.toMap()
+                )
+                analyze(singleFileInfo)
+            }
+
+            result.analyzer?.result shouldNotBeNull {
+                projects shouldHaveSize 90
```

Comment:
I think it would be valuable to make the fun test consistent with the ones for other package managers,
to use an expected output file which contains the expected ORT model. This would make it more obvious what the package manager outputs and protect from unintended breaking changes when changing the SPDX analyzer.